public/components/common/seach/queryBar.tsx

```diff
@@ -72,53 +72,64 @@ export function QueryBar(props: IQueryBarProps) {
     handleQuerySearch
   } = props;
 
-  const [inputValue, setInputValue] = useState('');
+  // State to keep track of current input
+  // Query contains the queries to send to backend, inputValue just keeps track of what's
+  // in the input bar at the time. This ensures that the query doesn't disappear when we
+  // switch tabs or hit enter.
+  const [inputValue, setInputValue] = useState(query[ RAW_QUERY ]);
 
   const splittedModel = inputValue.split(' ');
   const prefix = splittedModel[splittedModel.length - 1];
-  // const suggestions = commands.filter(
-  //   ({ label }) => label.startsWith(prefix) && prefix !== label && prefix !== ''
-  // );
 
   // Function to grab suggestions
   const getSuggestions = () => {
-    let suggest = commands;
-    // if (inputValue === "") {
-    //   console.log('test')
-    //   return suggest;
-    // }
-    if (splittedModel.length ===1) {
-      // if (prefix.startsWith("source=")) {
-      //   let indices = getIndices();
-      //   return indices;
-      // }
+    // First commands should either be search or source
+    // source should be followed by an available index
+    if (splittedModel.length === 1) {
+      if (prefix.startsWith("source=")) {
+        let str = prefix.replace("source=", "");
+        return getIndices(str);
+      }
       return firstCommand;
-      // return firstCommand.filter(
-      //   ( { label } ) => label.startsWith(prefix) && prefix !== label 
-      // );
     }
+    // TODO: (Grammar implementation) Get commands based on grammar from backend
+    // Contextual suggestions are hardcoded for now
     if (splittedModel.length > 1) {
-      // User just inputted a pipe
+      // Possible pipe commands
       if (splittedModel[splittedModel.length - 2] === "|"){
```

Comment:
if user input is `source = index| head 10`, then `|` will not be on its own when you split by space?

Reply:
forgot there could be spaces between `source` and `=`. i'm working on a more extensive grammar checker that is hardcoded for now and i'll remember to take this into account